packages/commonwealth/client/scripts/views/pages/notification_settings/index.tsx

```diff
@@ -47,6 +58,41 @@ const NotificationSettingsPage = () => {
     app.user.notifications.isLoaded.once('redraw', forceRerender);
   }, [app?.user.notifications, app.user.emailInterval]);
 
+  useEffect(() => {
+    const getTheSpace = async () => {
+      try {
+        // Fetch spaces for all snapshotIds concurrently
+        const spacePromises = snapshotIds.map(async (snapshotId) => {
+          try {
+            return await getSpace(snapshotId);
+          } catch (error) {
+            console.error(
+              `Error getting space for snapshotId ${snapshotId}:`,
+              error
+            );
+            return null; // You can handle the error as needed
+          }
+        });
+
+        // Wait for all space fetch requests to complete
+        const spaceArray = await Promise.all(spacePromises);
+
+        // Combine bundledSnapshotSubs with spaceArray
+        const snapshotsInfoArr = snapshotIds.map((snapshotId, index) => ({
+          snapshotId,
+          space: spaceArray[index],
+          subs: bundledSnapshotSubs[snapshotId],
+        }));
+
+        setSnapshotsInfo(snapshotsInfoArr);
+      } catch (error) {
+        console.error('Error fetching spaces:', error);
+      }
+    };
+
+    getTheSpace();
+  }, []);
```

Comment:
React Hook useEffect has missing dependencies: 'bundledSnapshotSubs' and 'snapshotIds'. Either include them or remove the dependency array

Reply:
I refactored this entirely. All the dependencies we're actually only used within this `useEffect` so i pulled them in.